packages/synapse-react-client/src/components/EntityFinder/tree/VirtualizedTree.tsx

```diff
@@ -523,27 +538,44 @@ export const VirtualizedTree = (props: VirtualizedTreeProps) => {
     // Because we update the root node with a copy at the end of this function, we can write to the node under update.
     async (node: Writable<EntityHeaderNode>) => {
       // Fetch the children of the node
-      const children = await SynapseClient.getEntityChildren(
-        {
-          parentId: node.id,
-          nextPageToken: node.childrenNextPageToken,
-          includeTypes: visibleTypes,
-        },
-        accessToken,
-      )
-
+      const entityChildrenRequest: EntityChildrenRequest = {
+        parentId: node.id,
+        nextPageToken: node.childrenNextPageToken,
+        includeTypes: visibleTypes,
+      }
+      const children = await queryClient.fetchQuery({
+        queryKey: keyFactory.getEntityChildrenQueryKey(
+          entityChildrenRequest,
+          false,
+        ),
+        queryFn: () => getEntityChildren(entityChildrenRequest, accessToken),
+      })
```

Comment:
Since this runs in a callback, we can't use `useQuery`, but we can still manually use/store query data using `queryClient`.

Reply:
Nice! If `useGetEntityChildren` was updated in the future, would this code need to be similarly updated? Just wondering whether that would be obvious to the dev making the change or whether it would be helpful to add a comment to remind us to update this code if `useGetEntityChildren` is updated

Reply:
This is a general issue, but I am not convinced comments would help very much.

If the `keyFactory.getEntityChildrenQueryKey` signature changes then we would catch it but that might not be the case.

One option I have thought a little bit about is that it might make sense to set up some data structure to bind all of our query keys and their corresponding query functions, and by convention only pull queries from that structure. Thought that might be awkward to use if the query functions rely on some resource (for example, an instance of a class generated by the OpenAPI spec that stores auth state and makes API calls).

Reply:
Thanks for the context, it does seem like a tricky problem to solve! Agreed that comments probably aren't the solution